content/en/blog/_posts/data-tbd/index.md

```diff
@@ -0,0 +1,313 @@
+for it are reserved. This also then tells the scheduler where in the cluster a
+ResourceClaim is available.
+
+ResourceClaims can get allocated as soon as they are created ("immediate
+allocation"), without considering which Pods will use them. The default is to
+delay allocation until a Pod gets scheduled which needs the ResourceClaim
+(i.e. "wait for first consumer").
+
+In that mode, the scheduler checks all ResourceClaims needed by a Pod and
+creates a PodScheduling object where it informs the resource drivers
+responsible for those ResourceClaims about nodes that the scheduler considers
+suitable for the Pod. The resource drivers respond by excluding nodes that
+don't have enough of the driver's resources left. Once the scheduler has that
+information, it selects one node and stores that choice in the PodScheduling
+object. The resource drivers then allocate their ResourceClaims so that the
+resources will be available on that node. Once that is complete, the Pod
+gets scheduled.
+
+As part of this process, ResourceClaims also get reserved for the
+Pod. Currently ResourceClaims can either be used exclusively by a single Pod or
+an unlimited number of Pods.
+
+One key feature is that Pods do not get scheduled to a node unless all of
+their resources are allocated and reserved. This avoids the scenario where a Pod
+gets scheduled onto one node and then cannot run there, which is bad because
+such a pending Pod also blocks all other resources like RAM or CPU that were
+set aside for it.
+
+## Limitations
+
+The scheduler plugin must be involved in scheduling Pods which use
+ResourceClaims. Bypassing the scheduler by setting the `nodeName` field leads
+to Pods that the kubelet refuses to start because the ResourceClaims are not
+reserved or not even allocated. It may be possible to [remove this
+limitation](https://github.com/kubernetes/kubernetes/issues/114005) in the
+future.
+
+## Writing a resource driver
+
+A dynamic resource allocation driver consists of two separate-but-coordinating
+components: a centralized controller and a daemonset of node-local kubelet
+plugins. Most of the work required by the centralized controller to coordinate
+with the scheduler can be handled by boilerplate code. Only the business logic
+required to actually allocate ResourceClaims against the ResourceClasses owned
+by the plugin needs to be customized. As such, the following package is
+provide by Kubernetes to include APIs for invoking this boilerplate code as
+well as a Driver interface that one can implement to provide their custom
+business logic:
+
+- [k8s.io/dynamic-resource-allocation/controller](https://github.com/kubernetes/dynamic-resource-allocation/tree/master/controller)
+
+Likewise, boilerplate code can be used to register the node-local plugin with
+the kubelet, as well as start a gRPC server to implement the kubelet plugin
+API. The following package is provided for this purpose:
```

Comment:
```suggestion
API. Kubernetes provides the following example code package for this purpose:
```
?

Reply:
I don't know if I'd really call it "example" code. The intention is that most (if not all) resource drivers would use this package to avoid having to reimplement this logic themselves.

Reply:
We're making a big assumption that people code in Golang, then.

Reply:
We could also say that we recommend it; does that help?

Reply:
True. I would definitely say is is recommended when writing resource drivers in go. Similar libraries could be created for other languages.

Reply:
Changed to:
> For drivers written in Go, the following package is recommended: